hfta/ops/batchnorm_test.py

```diff
@@ -55,6 +55,44 @@ def testcase_1d(
       except AssertionError as e:
         print(e)
 
+    # emulate "training" for the BN layer
+    [bn.train() for bn in batchNormal1d_array]
+    for i in range(train_test_steps):
+      if L == 0:
+        x_array = [torch.rand(N, C) for _ in range(B)]
+      else:
+        x_array = [torch.rand(N, C, L) for _ in range(B)]
+
+      _ = [batchNormal1d_array[b](x_array[b]) for b in range(B)]
+
+    batchNormal1d_fused.reset_parameters()
+    # snatch the "trained" BN layer
```

Comment:
Hmmm, there's neither backward passes nor parameter update steps, how is this `# emulate "training" for the BN layer`?

Reply:
I thought running forward pass already changes the internal state of BN layer ?
the weights and bias

Reply:
Yes, sorry, you are right, my bad :)
`running_mean`, `running_var` and `num_batches_tracked` are registered as buffers: https://github.com/UofT-EcoSystem/hfta/blob/main/hfta/ops/batchnorm.py#L47, and indeed they are not updated by the optimizers: https://discuss.pytorch.org/t/what-is-the-difference-between-register-buffer-and-register-parameter-of-nn-module/32723 . I misread them as `register_parameter()`